carbon-design-system/carbon
# fix(sketch): resolve icon sync artwork issues

Closes #5749
Closes #7712

This PR builds off the symbol name changes in #7648 to resolve issues with the icons sync command skewing artwork or not properly replacing outdated artwork

#### Changelog

**Changed**

- refactor `syncSymbol` and `syncIconSymbols` function signatures to use param object instead of a list of params. This helped with linting by triggering warnings with unused params while refactoring. It will also help in case new params need to be added in the future, since the order of the function params will be unimportant within the config object.

**Removed**

- `syncSymbolLayers` function
- `merge` function

#### Testing / Reviewing

- Confirm that existing artwork is updated to match the artwork from the library
- Confirm that new symbols are added to the library
- Confirm that local shared styles are preserved when updating the library

[carbon-elements.sketchplugin.zip](https://github.com/carbon-design-system/carbon/files/5906690/carbon-elements.sketchplugin.zip)
